app/utils/prereqCheck.js

```diff
@@ -0,0 +1,94 @@
+import path from 'path'
+import { execSync } from 'child_process'
+
+import arch from 'arch'
+import { dialog } from 'electron'
+
+import { getZenNodePath } from '../ZenNode'
+
+import { isOsx, isWindows, isLinux } from './platformUtils'
+
+export default prereqCheck
+
+const ERR_STATUS = 'ERR_STATUS'
+
+function prereqCheck() {
+  console.log('\n***** prereq check: start *****')
+  validateOs()
+  validateZ3()
+  validateMono()
+  console.log('***** prereq check: passed *****\n')
+}
+
+function validateOs() {
+  if (arch() !== 'x64') {
+    const msg = 'Zen-node can only run on 64 bit OS'
+    console.log(msg)
+    dialog.showErrorBox(
+      'Operating system not supported',
+      msg,
+    )
+    process.exit(1)
+  } else {
+    console.log('OS .64 version check passed')
+  }
+}
+
+function validateZ3() {
+  try {
+    const z3Path = getZ3PathForPlatform()
+    execSync(`${z3Path} --help`)
+    console.log('Z3 check passed')
+  } catch (err) {
+    dialog.showErrorBox(
+      'Z3 check failed',
+      `Failed to run z3\n${err.message}`,
+    )
+    process.exit(1)
+  }
+}
+
+function getZ3PathForPlatform() {
+  const pathToDir = path.join(getZenNodePath(), 'Release')
+  let filename
+  if (isWindows) { filename = 'z3.exe' }
+  if (isOsx) { filename = 'z3-osx' }
+  if (isLinux) { filename = 'z3-linux' }
+  return path.join(pathToDir, filename)
+}
+
+function validateMono() {
+  if (!isOsx()) {
+    console.log('mono version check passed (not needed for non OSX)')
+    return
+  }
+  const { status, msg } = getMonoStatus()
+  if (status === ERR_STATUS) {
+    console.error(msg)
+    dialog.showErrorBox(
+      'Mono check failed',
```

Comment:
this is never printed instead there is an error ENOENT, which usually mean that the mono version is not updated or that mono is not installed

Reply:
Thank you for reporting @Cremba , it works for me on ubuntu when I trigger the error deliberately by passing a wrong path.

Will look into it.

Reply:
[PR](https://github.com/zenprotocol/zenwallet/pull/82) attempted fix